File: 380-insert-delete-getrandom-o1/380-insert-delete-getrandom-o1.cpp
class RandomizedSet {
private:
    unordered_set<int> s;
    int size;
    
public:
    RandomizedSet() {
        size = 0;
    }
    
    bool insert(int val) {
        if(s.find(val) != s.end())
            return false;
        s.insert(val);
        size++;
        return true;
    }
    
    bool remove(int val) {
        if(s.find(val) == s.end())
            return false;
        s.erase(val);
        size--;
        return true;
    }
    
    int getRandom() {
        // srand(time(0));
        
        int pos = (rand()) % size;
        auto it = s.begin();
        advance(it, pos);
        return (*it);
    }
};

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet* obj = new RandomizedSet();
 * bool param_1 = obj->insert(val);
 * bool param_2 = obj->remove(val);
 * int param_3 = obj->getRandom();
 */

File: 380-insert-delete-getrandom-o1/380-insert-delete-getrandom-o1.py
import random
class RandomizedSet:
    def __init__(self):
        self.s=[]

    def insert(self, val: int) -> bool:
        if val in self.s:
            return False
        self.s.append(val)
        return True

    def remove(self, val: int) -> bool:
        if val in self.s:
            self.s.remove(val)
            return True
        return False
        

    def getRandom(self) -> int:
        return random.choice(self.s)
        